main/SAPI.c

```diff
@@ -205,15 +205,22 @@ static void sapi_read_post_data(void)
 		SG(request_info).post_entry = NULL;
 		if (!sapi_module.default_post_reader) {
 			/* no default reader ? */
-			SG(request_info).content_type_dup = NULL;
+			if (SG(request_info).content_type_dup) {
+				efree(SG(request_info).content_type_dup);
+				SG(request_info).content_type_dup = NULL;
```

Comment:
I'm not sure that we are supposed to free `SG(request_info).content_type_dup` in this case. If I understand this code correctly, `sapi_read_post_data()` is supposed to initialize `SG(request_info).content_type_dup` and ignore any previous value. A non-NULL `SG(request_info).content_type_dup` would be a freed pointer from a previous request.

Reply:
I see in `sapi_deactivate()` there is a `efree(SG(request_info).content_type_dup);` for non null `content_type_dup` without set it to NULL afterwards. Overall, an unfortunate coding style.

Reply:
So I had a closer look and the `sapi_deactivate` is not the only place that frees `SG(request_info).content_type_dup`.  I see also `sapi_handle_post` which is called by treat_data hook which I assume will be called later. It also sets it to NULL so it should never happen that `SG(request_info).content_type_dup` is not `NULL` here unless I missed something..? I think it doesn't really do any harm to add it but I don't see it as a fix in such case.